### PR TITLE
Migrate entity settings dialog

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Migrate entity settings dialog to `mwc-dialog` to match more info.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/entities/dialog-entity-editor.ts [hunk 1]
```diff
@@ -87,88 +83,89 @@ export class DialogEntityEditor extends LitElement {
...
+                  >
+                    <ha-svg-icon .path=${mdiTune}></ha-svg-icon>
+                  </mwc-icon-button>
                 `
-            : this._curTab === "tab-related"
-            ? html`
-                <paper-dialog-scrollable>
+              : ""}
+          </app-toolbar>
+          <paper-tabs
```
Comment (balloob) [inline src/panels/config/entities/dialog-entity-editor.ts:117]: One day we should kill this too :)

File: src/panels/config/entities/dialog-entity-editor.ts [hunk 2]
```diff
@@ -87,88 +83,89 @@ export class DialogEntityEditor extends LitElement {
...
+              `
+            )}
+            <paper-tab id="tab-related">
+              ${this.hass.localize("ui.dialogs.entity_registry.related")}
+            </paper-tab>
+          </paper-tabs>
+        </div>
+        <div class="wrapper">
+          ${cache(
+            this._curTab === "tab-settings"
```
Comment (balloob) [inline src/panels/config/entities/dialog-entity-editor.ts:138]: If logic gets this complicated, we could consider to extract it as a `renderTab` function and just do a bunch of `if (X) return Y`, might be more readable.

File: src/panels/config/entities/editor-tabs/settings/entity-settings-helper-tab.ts [hunk 3]
```diff
@@ -120,32 +117,30 @@ export class EntityRegistrySettingsHelper extends LitElement {
...
+        ${!this._componentLoaded
+          ? this.hass.localize(
+              "ui.dialogs.helper_settings.platform_not_loaded",
+              "platform",
+              this.entry.platform
+            )
+          : this._item === null
+          ? this.hass.localize("ui.dialogs.helper_settings.yaml_not_editable")
+          : html`
+              <div @value-changed=${this._valueChanged}>
```
Comment (balloob) [inline src/panels/config/entities/editor-tabs/settings/entity-settings-helper-tab.ts:131]: If we just care about capturing events, let's use a span so we don't accidentally do layout stuff.

**Review by balloob — APPROVED** [review]

File: src/panels/config/entities/dialog-entity-editor.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/config/entities/dialog-entity-editor.ts:117]: yeah, not sure if the side scrolling in `mwc` is fixed already

**Pull request merged**